tools/topology/sof-tgl-max98357a-rt5682.m4

```diff
@@ -97,8 +97,18 @@ define(DMIC_PIPELINE_KWD_ID, `12')
 define(DMIC_DAI_LINK_16k_ID, `2')
 # define pcm, pipeline and dai id
 define(KWD_PIPE_SCH_DEADLINE_US, 5000)
-# include the generic dmic with kwd
-include(`platform/intel/intel-generic-dmic-kwd.m4')
+
+# if RTNR is defined, define DMIC16KPROC as rtnr
+ifdef(`RTNR',
+`define(DMICPROC, rtnr)',
+`ifdef(`DMICPROC', , `define(DMICPROC, eq-iir-volume)')')
+
+ifdef(`RTNR',
+`define(DMIC16KPROC, rtnr)',
+`ifdef(`DMIC16KPROC', , `define(DMIC16KPROC, eq-iir-volume)')')
+
+# include the generic dmic if RTNR is defined, else include generic dmic with kwd
+include(ifdef(`RTNR', platform/intel/intel-generic-dmic.m4, platform/intel/intel-generic-dmic-kwd.m4))
```

Comment:
i missed this in my initial review, @MingJenTai this wont work. You simply can't remove hotwording. They both need to work in this config.

Reply:
@cujomalainey Ok I'll have a fix for that. Thanks.

Reply:
@cujomalainey I've created PR https://github.com/thesofproject/sof/pull/4988 for this. Thank you.